### PR TITLE
[#158285425] Implement payment from message preview and message detail screen

### PR DESCRIPTION
**Pull request by cloudify**

### CODE REVIEW

File: ts/utils/payment.ts [hunk 1]
```diff
@@ -8,3 +20,24 @@ export function formatPaymentAmount(amount: number): string {
   // We need to divide by 100 to get euro from euro-cents
   return I18n.toNumber(amount / 100, { precision: 2 });
 }
+
+/**
+ * Converts a NoticeNumber coming from a Message to an RptId needed by PagoPA
+ */
+export function getRptIdFromNoticeNumber(
```
Comment (cloudify) [inline ts/utils/payment.ts]: @fpersico @fgiobergia @gunzip @LsKoder  is this the right way to convert the data coming from a message to what is expected by PagoPA?

File: ts/components/messages/MessageComponent.tsx [hunk 2]
```diff
@@ -17,13 +21,14 @@ import {
...
 import { formatPaymentAmount } from "../../utils/payment";
+import { paymentRequestTransactionSummaryFromRptId } from "../../store/actions/wallet/payment";
 
 export type OwnProps = MessageDetailsByIdState &
   Readonly<{
     navigation: NavigationScreenProp<NavigationState>;
   }>;
 
-export type Props = OwnProps;
+export type Props = OwnProps & NavigationInjectedProps;
```
Comment (fpersico) [inline ts/components/messages/MessageComponent.tsx]: The component use connect
change NavigationInjectedProps to ReduxProps and
this.props.navigation.dispatch to this.props.dispatch

File: ts/utils/payment.ts (hunk 1 above)
Comment (fgiobergia) [reply ts/utils/payment.ts]: `RptId` [has been defined](https://github.com/teamdigitale/italia-ts-commons/blob/e395e75cf40fe37b362c4858f0279d75f26c20b1/src/pagopa.ts#L360) to contain the payment notice number and the organization fiscal code (the two fields that identify a payment). 

What you are using here is the payment notice number alone (`noticeNumber`), but you also need to get the organization fiscal code (of type [`OrganizationFiscalCode`](https://github.com/teamdigitale/italia-ts-commons/blob/e395e75cf40fe37b362c4858f0279d75f26c20b1/src/strings.ts#L143) in order to build a valid `RptId`. 

`RptIdFromString` is a utility function that converts a string in the format `${fiscalCode}${noticeNumber}` into an RptId (which is then passed to the proxy)

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#158285425](https://www.pivotaltracker.com/story/show/158285425): Come CIT quando apro il dettaglio di un messaggio di pagamento voglio poter iniziare il pagamento

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**